python/pfs/drp/stella/fitBroadbandSED.py

```diff
@@ -0,0 +1,135 @@
+from lsst.utils import getPackageDir
+
+import numpy
+import numpy.lib.recfunctions
+import astropy.io.fits
+
+import os
+
+
+class FitBroadbandSEDConfig(Config):
+    """Configuration for FitBroadbandSEDTask
+    """
+
+    fluxLibraryPath = Field(
+        dtype=str,
+        default=os.path.join(getPackageDir("drp_pfs_data"), "fluxCalibration",
+                             "syntheticFlux_withInterp_ps_HSCPS1GaiaSDSS.fits"),
+        doc="Synthetic photometry table"
+    )
+
+    filterAliases = DictField(
+        keytype=str, itemtype=str,
+        default={
+            "g": "HSCg", "r": "HSCr", "r2": "HSCr2", "i": "HSCi", "i2": "HSCi2",
+            "z": "HSCz", "y": "HSCy"
+        },
+        doc="Conversion table from AFW's filter names to those used by `fluxLibrary`"
+    )
+
+
+class FitBroadbandSEDTask(Task):
+    """Fit an observed SED with synthetic model SEDs
+    """
+
+    ConfigClass = FitBroadbandSEDConfig
+    _DefaultName = "fitBroadbandSED"
+
+    def runDataRef(self, dataRef):
+        """For each spectrum,
+        calculate probabilities of model SEDs matching the spectrum.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.ButlerDataRef`
+            Data reference for merged spectrum.
+
+        Returns
+        -------
+        broadbandPDF : `list` of `numpy.array` of `float`
+            ``broadbandPDF[iSpectrum][iSED]`` is the normalized probability
+            of the SED ``iSED`` matching the spectrum ``iSpectrum``.
+        """
+        pfsConfig = dataRef.get("pfsConfig")
+
+        pdfs = []
+        for filterNames, fiberFlux, fiberFluxErr \
+                in zip(pfsConfig.filterNames, pfsConfig.fiberFlux, pfsConfig.fiberFluxErr):
+            pdfs.append(self.run(filterNames, fiberFlux, fiberFluxErr))
+
+        return pdfs
+
+    def run(self, filterNames, fiberFlux, fiberFluxErr):
+        """Calculate a chi square and a probability for each model SED.
+
+        Parameters
+        ----------
+        filterNames : `list` of `str`
+            List of filters used to measure the fiber fluxes for each filter.
+            e.g. ``["g", "i"]``.
+        fiberFlux : `list` of `float`
+            Array of fiber fluxes for each fiber, in [nJy].
+        fiberFluxErr : `list` of `float`
+            Array of fiber flux errors for each fiber in [nJy].
+
+        Returns
+        -------
+        broadbandPDF : `numpy.array` of `float`
+            Array of normalized probabilities of the SED fitting.
+        """
+        if not (len(filterNames) == len(fiberFlux) == len(fiberFluxErr)):
+            raise ValueError("Lengths of arguments must be equal.")
+        if not filterNames:
+            nSEDs = len(self.fluxLibrary)
+            return numpy.full(shape=(nSEDs,), fill_value=1.0/nSEDs, dtype=float)
+
+        # Convert filter names.
+        filterNames = [self.config.filterAliases.get(f, f) for f in filterNames]
+
+        # fluxLibrary.shape = (nSEDs, nBands)
```

Comment:
This looks like code, and having commented-out code is a violation of the style guide, so I suggest just prefixing it with "Note:". Ditto below.